libraries/botbuilder-storage-mssql/src/mssqlStorage.ts

```diff
@@ -0,0 +1,105 @@
+import { Storage, StoreItem, StoreItems } from 'botbuilder-core';
+import { config as msconfig, ConnectionPool, NVarChar, IResult, Int } from 'mssql';
+import { MSSQLOptions } from './schema';
+
+/**
+ * @module @botbuildercommunity/storage-mssql
+ */
+
+function getConnectionPool(connection): ConnectionPool {
+    try {
+        return new ConnectionPool(connection);
+    }
+    catch(e) {
+        console.error(e);
+    }
+}
+
+export class MSSQLStorage implements Storage {
+    private options: MSSQLOptions;
+    private connection: msconfig;
+    public constructor(options: MSSQLOptions) {
+        this.options = options;
+        this.connection = {
+            user: options.dbuser,
+            password: options.dbpassword,
+            server: options.dbserver,
+            database: options.db
+        };
+    }
+    public async read(keys: string[]): Promise<StoreItems> { //Should this be an array of numbers?
+        if(keys == null || keys.length === 0) {
+            return Promise.resolve({});
+        }
+        const pool = await getConnectionPool(this.connection).connect();
+        try {
+            const result: IResult<any> = await pool.request()
+                .query(`SELECT
+                            id,
+                            data
+                        FROM ${ this.options.dbtable }
+                        WHERE id in (${ keys.join(',') })`); //Need to sanitize this.
+            const res = result.recordset.reduce((acc: StoreItem, record: any): StoreItem => {
+                acc[record.id] = acc[record.data];
+                return acc;
```

Comment:
`acc`? Could it be a more descriptive name.

Reply:
It's short for accumulator :smile:

Reply:
Changed to `storeItem`.